ctlearn/default_models/single_cnn.py

```diff
@@ -8,6 +8,7 @@ def single_cnn_model(data, model_params):
 
     # Load neural network model
     network_input = tf.keras.Input(shape=data.img_shape, name=f"images")
+    network_input_param = tf.keras.Input(shape=data.prm_shape, name=f"parameters")
```

Comment:
I think that should break if the parameter list is empty. Try to run the default model, i.e. `-d TRN`,  or comment out the parameter list in your config file with this version. Also, try to test the scenario where you only have the parameter list and no images.

Reply:
Yes removing the parameter list will cause an issue. We could have a check here if the parameter list is non-empty, and use a flag so that the parametric model lines only execute if required. Does that seem fine? And for the case when there is only a parameter list and no images, we would ideally want a separate parameter model similar to the previous PR I had, right?

Reply:
Your suggestion with a flag sounds good! I think `data.img_shape` and/or `data.prm_shape` should be none if they are not selected. that could even be your flag. Caution in predict mode we normally select also the parameters to store them to disk in the prediction (Data Level 2) file. Therefore your flag for the parameter should also contain a check the mode, i.e. checking if training is selected.

Reply:
Okay I will keep that in mind and update this. Thanks!

Reply:
I have added a flag_prm to toggle between the single CNN model with or without parameters. It is working fine in both cases. Do have a look.